platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/internal/PluginUsePluginServiceRegistry.java

```diff
@@ -96,14 +96,34 @@ AutoAppliedPluginHandler createAutoAppliedPluginHandler(List<AutoAppliedPluginRe
         }
 
         ClientInjectedClasspathPluginResolver createInjectedClassPathPluginResolver(
-            ClassLoaderScopeRegistry classLoaderScopeRegistry, PluginInspector pluginInspector,
-            InjectedPluginClasspath injectedPluginClasspath, CachedClasspathTransformer classpathTransformer,
+            FileResolver fileResolver,
+            DependencyManagementServices dependencyManagementServices,
+            DependencyMetaDataProvider dependencyMetaDataProvider,
+            ClassLoaderScopeRegistry classLoaderScopeRegistry,
+            PluginInspector pluginInspector,
+            InjectedPluginClasspath injectedPluginClasspath,
+            ScriptClassPathResolver scriptClassPathResolver,
+            FileCollectionFactory fileCollectionFactory,
             InjectedClasspathInstrumentationStrategy instrumentationStrategy
         ) {
             if (injectedPluginClasspath.getClasspath().isEmpty()) {
                 return ClientInjectedClasspathPluginResolver.EMPTY;
             }
-            return new DefaultInjectedClasspathPluginResolver(classLoaderScopeRegistry.getCoreAndPluginsScope(), classpathTransformer, pluginInspector, injectedPluginClasspath.getClasspath(), instrumentationStrategy);
+            Factory<DependencyResolutionServices> dependencyResolutionServicesFactory = makeDependencyResolutionServicesFactory(
```

Comment:
Instead of creating a _new_ DependencyResolutionServicesInstance, it looks like we can inject a `PluginDependencyResolutionServices` instance into this method. Then we can expose the service we need on that type and pass it into the `DefaultInjectedClasspathPluginResolver` constructor

Reply:
That was my plan at first, but it seem we need to mutate some of its services to register artifact transform, e.g. we need to call `scriptClassPathResolver.prepareDependencyHandler(dependencies);`. If this has no negative effect then I can change it.

Reply:
Ah I see. I see two options

1. Keep it the way it is. Run a perf test to make sure creating new services instances doesn't slow things down
2. Use the same instance everywhere and register the transforms when creating `PluginDependencyResolutionServices`

Reply:
I would go with 1., since TestKit plugins anyway uses isolated classpath already. 

I think there should not be any performance regressions since this path is triggered only via TestKit, in other cases we return `ClientInjectedClasspathPluginResolver.EMPTY`. But just in case I will run performance tests.

Reply:
Yeah in the future we should add some ad-hoc way of running a resolution with a one-off set of artifact transforms. Something like this will come in time with some of the dependency management work we're doing